File: core/src/dc/targetman/mechanics/VitalLimbsSystem.java
package dc.targetman.mechanics;

import java.util.List;

import dc.targetman.epf.parts.VitalLimbsPart;
import dclib.epf.Entity;
import dclib.epf.EntityManager;
import dclib.epf.EntitySystem;
import dclib.physics.limb.Limb;
import dclib.physics.limb.LimbUtils;

public final class VitalLimbsSystem extends EntitySystem {

	private final EntityManager entityManager;

	public VitalLimbsSystem(final EntityManager entityManager) {
		super(entityManager);
		this.entityManager = entityManager;
	}

	@Override
	protected final void update(final float delta, final Entity entity) {
		VitalLimbsPart vitalLimbsPart = entity.tryGet(VitalLimbsPart.class);
		if (vitalLimbsPart != null) {
			List<Entity> entities = entityManager.getAll();
			for (Limb limb : vitalLimbsPart.getVitalLimbs()) {
				if (LimbUtils.INSTANCE.findEntity(entities, limb) == null) {
					entityManager.remove(entity);
					break;
				}
			}
		}
	}

}

File: core/src/main/dc/targetman/mechanics/VitalLimbsSystem.kt
package dc.targetman.mechanics

import dc.targetman.epf.parts.SkeletonPart
import dc.targetman.epf.parts.VitalLimbsPart
import dclib.epf.Entity
import dclib.epf.EntityManager
import dclib.epf.EntitySystem

class VitalLimbsSystem(private val entityManager: EntityManager) : EntitySystem(entityManager) {
    override fun update(delta: Float, entity: Entity) {
        val vitalLimbsPart = entity.tryGet(VitalLimbsPart::class)
        if (vitalLimbsPart != null) {
            val skeletonPart = entity[SkeletonPart::class]
            val isVitalLimbDead = vitalLimbsPart.limbNames.any { !skeletonPart[it].isActive }
            if (isVitalLimbDead) {
                entityManager.remove(entity)
            }
        }
    }
}